### ISSUE TITLE
ffmpeg.input.cpp.o failed

### ISSUE THREAD

**Issue by gree303** [OpenImageIO/oiio#1565]
### Problem
I've tried to compile oiio-release & oiio-master and wasn't able to fix the following error:
`[ 11%] Building CXX object src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/ffmpeg.imageio/ffmpeginput.cpp.o
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp: In member function ‘virtual bool OpenImageIO::v1_7::FFmpegInput::open(const string&, OpenImageIO::v1_7::ImageSpec&)’:
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp:393:14: error: ‘AV_PIX_FMT_YUV420P12BE’ was not declared in this scope
         case AV_PIX_FMT_YUV420P12BE:
              ^
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp:394:14: error: ‘AV_PIX_FMT_YUV420P12LE’ was not declared in this scope
         case AV_PIX_FMT_YUV420P12LE:
              ^
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp:395:14: error: ‘AV_PIX_FMT_YUV422P12BE’ was not declared in this scope
         case AV_PIX_FMT_YUV422P12BE:
              ^
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp:396:14: error: ‘AV_PIX_FMT_YUV422P12LE’ was not declared in this scope
         case AV_PIX_FMT_YUV422P12LE:
              ^
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp:397:14: error: ‘AV_PIX_FMT_YUV444P12BE’ was not declared in this scope
         case AV_PIX_FMT_YUV444P12BE:
              ^
/home/user/Downloads/oiio-release/src/ffmpeg.imageio/ffmpeginput.cpp:398:14: error: ‘AV_PIX_FMT_YUV444P12LE’ was not declared in this scope
         case AV_PIX_FMT_YUV444P12LE:
              ^
src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/build.make:813: recipe for target 'src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/ffmpeg.imageio/ffmpeginput.cpp.o' failed
make[3]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/ffmpeg.imageio/ffmpeginput.cpp.o] Error 1
make[3]: Leaving directory '/home/user/Downloads/oiio-release/build/linux64'
CMakeFiles/Makefile2:1926: recipe for target 'src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/all' failed
make[2]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/all] Error 2
make[2]: Leaving directory '/home/user/Downloads/oiio-release/build/linux64'
Makefile:147: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/user/Downloads/oiio-release/build/linux64'
Makefile:352: recipe for target 'cmake' failed
make: *** [cmake] Error 2
`
## SYSTEM
Bunsenlabs / Debian 8

### STEPS
- downloaded oiio-release + oiio-master
- Installed required packages, like 'cmake', fixed libGL.so error (missing symlink)
- make
- resulting error msg at 11% (see output above)

### REASON
Need to get a newer version of oiio up running. This way blender should be able to work with RAW DNG files.
Blender DNG+CR2 patch from sambler: [blender patch](https://gist.github.com/sambler/05afb5d61f8a16b5a58dd35cea0ca927)
[sambler's answer](http://blender.stackexchange.com/questions/57208/how-to-develop-a-vse-file-format-importer-for-dng-cr2-image-files/57245#57245)

### POSSIBLE WORKAROUND (not tested and not sure if possible)
- tried to remove the ffmpeg from the installer. I think blender uses ffmpeg in a direct way. Not sure if it will work out. Wasn't able to remove it entirely and gave up at some point...
Any clue how to get this right?

**Comment by gree303** [comment]
SOLVED:
add these lines to repositories + upgrade ffmpeg:
`deb http://www.deb-multimedia.org jessie main non-free
deb-src http://www.deb-multimedia.org jessie main non-free`

**Comment by lgritz** [comment]
Can you explain a bit more?

Do you know which versions are known to work and which are known to break?

**Comment by gree303** [comment]
...sure.
To reproduce the error you'd need Debian8/Bunsenlabs. I've followed the debug output from "make debug". It will tell you which lib is missing; install them via Synaptic. But this won't work for ffmpeg like below.
The following version did not work for "make" build process:
ffmpeg_3.2-2~bpo8+2_amd64
I've tried it with the oiio-master + oiio-release repositories.

To fix it use the answer from above.
Still WIP; might open another issue i ran into with openexr. Currently reproducing the issue...I'll let you know asap.
